File: Shantanu_Mamgain/Two-Pointer-Leetcode/Reverse-Prefix-of-Word.java
class Solution {
    public String reversePrefix(String word, char ch) {
        int end = word.indexOf(ch);
        if(end == -1){
            return word;
        }        

        StringBuilder str = new StringBuilder(word.substring(0, end+1));
        str.reverse();
        str.append(word.substring(end + 1));

        return str.toString();
    }
}

File: Lakshay Garg/Leetcode/Two_Pointers_Leetcode/Reverse_Prefix_of_Word.cpp
class Solution {
public:
    string reversePrefix(string w, char ch) {
        int pos=-1;
        for(int i=0;i<w.size();i++){
            if(w[i]==ch){
                pos=i;
                break;
            }
        }
        reverse(w.begin(),w.begin()+pos+1);
        return w;
    }
};
